phoible/templates/phoible.mako

```diff
@@ -1,10 +1,39 @@
 <%inherit file="app.mako"/>
 
 <%block name="brand">
-    <a class="brand" style="padding-top: 7px; padding-bottom: 5px;" href="${request.route_url('dataset')}" title="${request.dataset.name}">
+    <a class="brand" style="padding-top: 7px; padding-bottom: 5px;" href="${request.route_url('dataset')}"
+       title="${request.dataset.name}">
         ##[ˈfɔɪ.bl̴]
         <img height="25" src="${request.static_url('phoible:static/phoible_icon.png')}"/>
     </a>
 </%block>
 
 ${next.body()}
+
+<script>
+    // Add charissil CSS head to all views
+    let charissil = "${request.static_url('clld:web/static/css/charissil.css')}";
+    document.querySelector('head').innerHTML += '<link rel="stylesheet" href="' + charissil + '" type="text/css"/>';
+
+    // Append Menu Items
+    let menuUl = document.querySelector('#menuitem_sources').parentElement;
+    let home_li = document.querySelector('#menuitem_dataset');
+    let navbar_div = document.querySelector('#contextnavbar');
+
+    let li1 = document.createElement('li');
+    let li2 = document.createElement('li');
+    li1.innerHTML = '<a href="/conventions" title="Conventions">Conventions</a>';
+    li2.innerHTML = '<a href="/faq" title="FAQ">FAQ</a>';
+        % if hasattr(context.get('request'), 'conventions'):
+            li1.classList.add('active');
+            home_li.classList.remove('active');
+            navbar_div.remove();
+        % endif
+        % if hasattr(context.get('request'), 'faq'):
+            li2.classList.add('active');
+            home_li.classList.remove('active');
+            navbar_div.remove();
+        % endif
+    menuUl.append(li1);
+    menuUl.append(li2);
+</script>
```

Comment:
isn't this part done in `__init__` now?

Reply:
I think __init__.py only adds routes to the CLLD web app. This is the script to append menu navigation items to every view. Initially, it is done through the [home_comp.mako](https://github.com/ltxom/phoible/commit/9ede1aa1d5a9cacf2059c60f2ff7549eb15f9cb1#diff-ec677a5f3c712eae55bac11eab5ac3a40cb255c4f694dcdf7eb0dbdb0382a478R14), but it only shows the "Conventions" and "FAQ" pages under the right menu bar on the home page.

Reply:
Yes, the context menu (i.e. the secondary menu) is only shown on the home page. This is intended, though, to leave the screen to non-navigational content. I don't have a very strong opinion here, though. If you think these links should appear on each page, feel free to leave it as is.

Reply:
I would indeed like the Conventions and FAQ links to be part of the main nav / appear on every page.  I'm just surprised that injecting the links into HTML this way is the "right" way to do it.  I don't have a strong grasp on the detailed workings of CLLD / mako, but isn't there a way to have a list of "static" pages that get "built" alongside the dynamic ones?

Reply:
So, inserting links in the main menu is demonstrated here: https://github.com/clld/hindukush/blob/2c73150d83b13284b4d834f71eece53f75c76318/hindukush/__init__.py#L61-L68
The signature can be seen best here: https://github.com/clld/hindukush/blob/2c73150d83b13284b4d834f71eece53f75c76318/hindukush/__init__.py#L65

You pass a tuple (Element ID, func), where func takes two arguments (request and context) and returns a pair (URL, link label).

Reply:
Thank you for the method. I applied it and the Conventions and FAQ links are shown properly. I notice the secondary menu (with links of Legal, Download, About, etc.) is still shown under these two pages. Is there a way in the CLLD framework to not render it? Or is it good to use DOM in js to remove the <div> after the page is loaded?

Reply:
If a page template inherits from `phoible.mako` rather than from `home_comp.mako`, the context menu will not be displayed.